### DISCUSSION TITLE
Data for Ontario, Canada

### DISCUSSION THREAD

**Discussion by nataliadgepi** [neherlab/covid19_scenarios#658] [General]
Thank you for such an amazing tool. I have used your model for a local hospital in Toronto and it has been widely used to predict what is going on at the local level. In order to model future scenarios, I was wondering if it would be possible to upload the Ontario data now that it is available. I wish I knew how but am an Epi and don't have the coding knowledge to parse the data. https://data.ontario.ca/dataset?keywords_en=COVID-19

I also wondered if you had any information on whether your mitigations were working better for local scenarios as a cumulative sum of reduction of transmission over time (from x date to y date from 20-30% reduction, non overlapping time intervals) vs different interventions (school closures, reduction from 20-30% with overlapping time intervals).

**Comment by nnoll** [comment]
It's really nice to hear that you've found utility with the tool. Your use was exactly what we had envisioned when building it! I'll start work on an Ontario parser and post here when it's completed.

Would you mind elaborating on your second question a bit? When mitigation intervals overlap, the efficacy of each are combined multiplicatively in the overlapping region. For example, if you have two mitigations overlapping, both with 50% reduction, then in the region in between will correspond to a 25% reduction. If you ask if we have tested which recapitulates reality more accurately, we are still in the prototype phase of fitting to mitigation efficacies to real data so I can't give a confident answer here.

**Reply by nataliadgepi** [reply]
Thank you so much. A question related to the first, 1) is there a way I can upload my hospital data to calibrate without sharing it (privacy issues). Right now I'm calibrating the model using provincial data and then scaling it down to the hospital level. I guess this can't be done online/through the internet but are there any thoughts of making this useful offline?

My second question was answered by the latter half of your answer, which type of mitigation is fitting more accurately.

Another question: 
2) Is it possible to include the severity table used in the pdf output, this is particularly useful to communicate and keep track of specific changes made in the model?
3) Is it possible to output the R(t) values? I love this graph and would love to be able to reproduce it. Would it be possible to give upper and lower bounds for R(t) for the future (say 2/3 waves)?

Once again, thank you for all your hard work. Absolutely incredible. Would be nice to write this up and show how it was used at a local level.

**Reply by nataliadgepi** [reply]
Finally, if you can include an option to calculate PPE based on model predictions that would be amazing. The CHIME calculator is here: https://code-for-philly.gitbook.io/chime/ppe-calculator/introduction-to-ppe

**Reply by nnoll** [reply]
We just merged a [pull request](https://github.com/neherlab/covid19_scenarios/pull/381) today that allows for import of user case data. After we polish it, it should make it to the production site. Hopefully it should be up soon!

2) We can certainly include this table. It should probably be there. We'll look at adding it for our next release.
3) We could include an additional column in our export .tsv files. Would this work for you?

We had brief discussion with the CHIME group to include the calculator with our model. I'll look into it and see how hard it would be to include this option.

**Reply by nataliadgepi** [reply]
That is spectacular and I can't wait for the next release. Are you considering any mobility data to predict future waves/scenarios (apple/google). Have you seen modelingcovid.com from stripe? 

Yes, adding the R(t) would be very useful, that would be spectacular ! We could use it to see how mitigations affected R in the past, and in the future to see what effect mitigations may have on future waves. 

There is no real rush for PPE at the moment although I could see this coming up again in future waves.

**Comment by nataliadgepi** [comment]
Hi @nnoll, 

Ontario has updated their dataset and if it could be uploaded that would really be helpful to be able to calibrate the Ontario model for covid: https://data.ontario.ca/dataset/status-of-covid-19-cases-in-ontario/resource/ed270bb8-340b-41f9-a7c6-e8ef587e6d11, it contains hospitalizations as well as ICU admissions. Please let me know if there is any way that I can help.

As always, Thank you!!!

**Reply by ivan-aksamentov** [reply]
Hi @nataliadgepi,

We will  incorporate the data eventually, but if you want it right now, you can try to use our shiny new case counts import feature. It is not released publicly yet, but is accessible on our internal testing build here:
https://master.covid19-scenarios.org/
(this version builds from `master` branch of the git repository on every change of the source code)

There you can click on  "or import your own data" link under "Confirmed Cases" dropdown
![Screenshot from 2020-05-25 11-14-40](https://user-images.githubusercontent.com/9403403/82798568-f7454e80-9e78-11ea-8245-3b35e40e154b.png)

and upload a [TSV](https://en.wikipedia.org/wiki/Tab-separated_values) or CSV file containing  the following columns:
 - time
 - cases
 - deaths
 - hospitalized
 - icu
 - recovered

"time" and "cases" are mandatory, the remaining are optional (you still need to have all the column headers/names, even if there is no data in the column)

For example, here is our current file for Ontario:
https://github.com/neherlab/covid19_scenarios/blob/master/data/case-counts/canada/CAN-Ontario.tsv
(use "Raw" button to see the plain text version of the file, which you can save with Ctrl+S)
You can for, example, open this file in Excel, add the necessary data, export as TSV file from Excel, and try to import it into the app.

Note that the case counts import feature might still be incomplete and there might be bugs, but we'd appreciate your early feedback. If you notice anything strange, please let me know. You can also attach a screenshot and the file you are trying to upload (here on GitHub you can drag and drop or paste files into the place in the message where you want them to appear uploaded).

I will be glad to help if you need anything.

**Reply by ivan-aksamentov** [reply]
@nataliadgepi,

In pull request #705 I implemented a parser for the Ontario data you mentioned above. Feel free to review the resulting TSV case counts data:
https://github.com/neherlab/covid19_scenarios/pull/705/files#diff-46557f9c6e1176ec307cd1cac209b556
Here red are the old (deleted) rows and green are the new (added) rows. As you see the numbers are a bit different, but not too much.

This is the end result in a prettier form:
https://github.com/neherlab/covid19_scenarios/blob/40c3b72a6b906b9c4e86324f4a2e4390fd6997c5/data/case-counts/canada/CAN-Ontario.tsv
It's the same format and you can also download, modify and import this file into the app, if needed.

And here is a testing version of the app for this particular pull request, including the new Ontario TSV data linked above:
https://covid19-scenarios-git-feat-data-canada-ontario.covid19-scenarios.now.sh/
Note that the default parameters (scenario section) are not yet fit for the new data, they are still from the old data.

If you are available, please take a look and let me know if it makes sense.

Both Richard and Nick are not available for a few days, but will review  the code changes later this week and then we can re-fit the params and, hopefully, release.

**Reply by nataliadgepi** [reply]
I will look at this tonight. This is really great thank you! I can add some more (earlier) data for ICU counts. Thank you so much!
